rmgpy/data/thermo.py

```diff
@@ -659,7 +662,10 @@ def load_entry(self,
         Note that these argument names are retained for backward compatibility.
         """
 
-        molecule = Molecule().from_adjacency_list(molecule)
+        try:
+            molecule = Molecule().from_adjacency_list(molecule)
+        except TypeError:
+            molecule = Fragment().from_adjacency_list(molecule)
```

Comment:
Minor: instead of a try/except, can we do `if isinstance(molecule, Molecule): ...   elif isinstance(molecule, Fragment): ...   else: raise TypeError(...)`

Reply:
This is my fault, I misunderstood what `molecule` was (i.e. it is actually an adjacency list that you can't do isinstance checks on). Sorry for the mixup here.